src/backend/BackendInterface.h

```diff
@@ -7,8 +7,13 @@
 #include <backend/Types.h>
 #include <thread>
 #include <type_traits>
+
 namespace Backend {
 
+/**
```

Comment:
If we are using doxygen, may as well use the full capabilities of the format:
```
/**
 * @brief Responsible for throwing an error when the database read time limit is exceeded.
 * 
 * Longer description if needed here...
 */
```

some example for a template function:
```
/**
 * @brief Sums two values
 * 
 * This function sums two values of specified `Type`.
 * Note that it's a requirement for `Type` to support the `+` operation.
 * 
 * @tparam Type The type of data
 * @param a The first value
 * @param b The second value
 * @return Type The sum of `a` and `b` 
 */
template<typename Type>
inline Type sum(Type a, Type b) const { return a + b; } 
```

i'd also pick one style and use it everywhere. so either the above or 
```
/// @brief ...
///
/// Text
/// @return ...
```

what i personally like to do is for short one-line doc-comments i do this:
```
class A {
    int member_;    /*! @brief Describe this member */
};
```
Because we have 80 character limit for clang-format.. the above might not look good. In which case it's also ok to put it above the `member_`.

To appreciate how the documentation looks you really should setup doxygen and actually generate docs from your code. Just experiment with it and see how it looks and what is better or worse in your opinion.

Reply:
yes lets use the full capabilities of doxygen

Reply:
This now looks better @officialfrancismendoza! I'd still like to see explanations for the return, tparam, etc. in some places.

Also, it seems like in the final output (html) this file was not included somehow. Please check doxygen configuration and make sure this file is part of the resulting html doc :+1: